emiproc/inventories/cams_reg_aq.py

```diff
@@ -0,0 +1,161 @@
+        year: int = 2022,
+        substances_mapping: dict[str, str] = {
+            "nox": "NOx",
+            "co": "CO",
+            "ch4": "CH4",
+            "nmvoc": "VOC",
+            "sox": "SO2",
+            "nh3": "NH3",
+            "pm2_5": "PM25",
+            "pm10":"PM10"
+        },
+        categories_mapping: dict[str, str] = {
+            "A_PublicPower": "A",
+            "B_Industry": "B",
+            "C_OtherStationaryComb": "C",
+            "D_Fugitives": "D",
+            "E_Solvents": "E",
+            "F_RoadTransport": "F",
+            "G_Shipping": "G",
+            "H_Aviation": "H",
+            "I_OffRoad": "I",
+            "J_Waste": "J",
+            "K_AgriLivestock": "K",
+            "L_AgriOther": "L"
+        },
+        substances_mapping_profiles: dict[str, str] = {
+            "nox": "NOx",
+            "co": "CO",
+            "ch4": "CH4",
+            "nmvoc": "VOC",
+            "so2": "SO2",
+            "nh3": "NH3",
+            "pm2_5": "PM25",
+            "pm10":"PM10"
+        },
+    ):
+        
+        super().__init__()
+
+        filename_pattern = fr"CAMS-REG-ANT_EUR_0\.05x0\.1_anthro_(?P<substance>\w+)_v6\.1-Ref2_yearly_{year}\.nc"
+
+        nc_dir = Path(nc_dir)
+        if not nc_dir.is_dir():
+                raise FileNotFoundError(
+                    f"Profiles directory {nc_dir} is not a directory."
+                )
+        nc_files = [f for f in nc_dir.iterdir() if f.is_file()]
+
+        if profiles_dir is None:
+            profiles_dir = Path(nc_dir)
+        else:
+            profiles_dir = Path(profiles_dir)
+            if not profiles_dir.is_dir():
+                raise FileNotFoundError(
+                    f"Profiles directory {profiles_dir} is not a directory."
+                )
+        
+        # Read the vertical and temporal profile files
+        v_profiles, v_profiles_indexes = read_vertical_profiles(profiles_dir)
+
+        t_profiles, t_profiles_indexes = read_temporal_profiles(
+            profiles_dir,
+            profile_csv_kwargs={
+                "encoding": "latin",
+            },
+        )
+        # Rename substances in profiles according to dictionary
+        if "substance" in t_profiles_indexes.dims:
+            t_profiles_indexes = t_profiles_indexes.assign_coords(
+                substance=[
+                    substances_mapping_profiles[name] for name in t_profiles_indexes['substance'].values
+                ]
+            )
+        if "substance" in v_profiles_indexes.dims:
+            v_profiles_indexes = v_profiles_indexes.assign_coords(
+                substance=[
+                    substances_mapping_profiles[name] for name in v_profiles_indexes['substance'].values
+                ]
+            )
+
+        # Read in emission data
+        inv_data = {}
+
+        for nc_file in nc_files:
+            if nc_file.suffix == '.nc' and re.match(filename_pattern, nc_file.name):
```

Comment:
Here I would move this condition above, where the nc_files list is created, so that the files selection is only done at one place